api/http/handler/edgestacks/edgestack_status_update.go

```diff
@@ -95,16 +97,16 @@ func (handler *Handler) updateEdgeStackStatus(tx dataservices.DataStoreTx, r *ht
 			return nil, nil
 		}
 
-		return nil, err
+		return nil, utils.NewEdgeError(strconv.Itoa(int(payload.EndpointID)), err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("unable to retrieve Edge stack from the database: %w. Environment ID: %d", err, payload.EndpointID)
```

Reply:
Always try to decorate the error using the fmt.Errorf pattern with more information about the context of the error.